src/xopen/__init__.py

```diff
@@ -898,8 +910,11 @@ def xopen(  # noqa: C901  # The function is complex, but readable.
     if mode not in ("rt", "rb", "wt", "wb", "at", "ab"):
         raise ValueError("Mode '{}' not supported".format(mode))
     binary_mode = mode[0] + "b"
+    # if file handle is passed: get name
+    file = filename
```

Comment:
I cannot add a review comment to the function signature above because it is too far away, so I’m writing it here: The `filename` parameter should be renamed to `file_or_path`.